qemu/tests/ovs_quit.py

```diff
@@ -24,33 +22,33 @@ def run(test, params, env):
 
     def get_ovs_ports(ovs):
         cmd = "ovs-vsctl list-ports %s" % ovs
-        return set(utils.system_output(cmd).splitlines())
+        return set(process.system_output(cmd).splitlines())
 
     os_dep.command("ovs-vsctl")
     netdst = params.get("netdst")
-    if netdst not in utils.system_output("ovs-vsctl list-br"):
-        raise error.TestError("%s isn't is openvswith bridge" % netdst)
+    if netdst not in process.system_output("ovs-vsctl list-br"):
+        test.error("%s isn't is openvswith bridge" % netdst)
 
     deps_dir = data_dir.get_deps_dir("ovs")
-    nic_script = utils_misc.get_path(deps_dir, params["nic_script"])
-    nic_downscript = utils_misc.get_path(deps_dir, params["nic_downscript"])
-    params["nic_script"] = nic_script
-    params["nic_downscript"] = nic_downscript
```

Comment:
I am wondering if the case still make sense after removed these params. I think avocado-vt could clean up the ovs port after QEMU quitted (not quite sure about this). Can you please have a check?

Reply:
OVS will clean the port, not avocado.

See the log below:
04:40:45 INFO | [qemu output] qemu-kvm: -device virtio-pci,mac=9a:ab:ac:ad:ae:af,id=idJz0CpZ,vectors=4,netdev=idYbHwdS,bus=pci.0,addr=0x5: Parameter 'driver' expects non-abstract device type
04:40:45 INFO | [qemu output] (Process terminated with status 1)
04:40:45 INFO | Created qemu process with parent PID 22669
04:40:45 INFO | Running 'ovs-vsctl list-ports ovs0'
04:40:45 DEBUG| [stdout] eno1
04:40:45 INFO | Command 'ovs-vsctl list-ports ovs0' finished with 0 after 0.00382709503174s


the output of 'ovs-vsctl list-ports ovs0' is 'eno1'. No more new created tap for guests.

Reply:
Updated other parts, please help to check again. Thanks.

Reply:
The truth is, before this change, QEMU used `downscript` to remove the given ports from the bridge device.
```
$ cat qemu-ifdown 
#!/bin/sh

# replace 'switch' openvswith bridge name in your system
switch="switch"
/sbin/ifconfig $1 0.0.0.0 down
/usr/bin/ovs-vsctl del-port ${switch} $1  <-- the ovs ports would be deleted at here
```
some buggy versions will not do this correctly, so the ovs ports would result in leak, see https://bugzilla.redhat.com/show_bug.cgi?id=884590 and https://bugzilla.redhat.com/show_bug.cgi?id=916114

After this change, now the tap device will be managed by avocado-vt because using fd mechanism, those ovs ports will be deleted by avocado-vt (in your case the path would be `qemu_vm.VM.create()` -> `qemu_vm.VM.destory()` -> `qemu_vm.VM._cleanup()`), for details please have a look at https://github.com/avocado-framework/avocado-vt/commit/9e75aa30fc071865f2fa63308f34c1c9341edade .

I just had a test on my side, it did work as I expected.
* default:
```
# ovs-vsctl show
f83262c3-cadb-44ff-bae3-5ac3558af7ae
    Bridge test
        Port test
            Interface test
                type: internal
    ovs_version: "2.0.0"

# run 'ovs_quit' with your patch ..

# ovs-vsctl show
f83262c3-cadb-44ff-bae3-5ac3558af7ae
    Bridge test
        Port test
            Interface test
                type: internal
    ovs_version: "2.0.0"
```
* without deleting ovs ports in avocado-vt:
```patch
diff --git a/virttest/qemu_vm.py b/virttest/qemu_vm.py
index 2ce7583..c54fbdf 100644
--- a/virttest/qemu_vm.py
+++ b/virttest/qemu_vm.py
@@ -3125,7 +3125,8 @@ class VM(virt_vm.BaseVM):
             elif nic.ifname and nic.ifname not in utils_net.get_net_if():
                 _, br_name = utils_net.find_current_bridge(nic.ifname)
                 if br_name == nic.netdst:
-                    utils_net.del_from_bridge(nic.ifname, nic.netdst)
+                    #utils_net.del_from_bridge(nic.ifname, nic.netdst)
+                    pass
 
     def destroy(self, gracefully=True, free_mac_addresses=True):
         """
```
result:
```
# ovs-vsctl show
f83262c3-cadb-44ff-bae3-5ac3558af7ae
    Bridge test
        Port test
            Interface test
                type: internal
    ovs_version: "2.0.0"

# run 'ovs_quit' with your patch ..

# ovs-vsctl show
f83262c3-cadb-44ff-bae3-5ac3558af7ae
    Bridge test
        Port test
            Interface test
                type: internal
        Port "t0-TiWvxt"                   <--
            Interface "t0-TiWvxt"
    ovs_version: "2.0.0"
```

So please check it.